backend/src/schema/types/type/User.gql

```diff
@@ -97,22 +97,22 @@ type User {
   contributions: [Post]! @relation(name: "WROTE", direction: "OUT")
   contributionsCount: Int! @cypher(
   statement: """
-  MATCH (this)-[: WROTE]->(r: Post)
+  MATCH (this)-[:WROTE]->(r:Post)
   WHERE NOT r.deleted = true AND NOT r.disabled = true
   RETURN COUNT(r)
   """
   )
 
   comments: [Comment]! @relation(name: "WROTE", direction: "OUT")
-  commentedCount: Int! @cypher(statement: "MATCH (this)-[: WROTE]->(: Comment)-[: COMMENTS]->(p: Post) WHERE NOT p.deleted = true AND NOT p.disabled = true RETURN COUNT(DISTINCT(p))")
+  commentedCount: Int! @cypher(statement: "MATCH (this)-[:WROTE]->(:Comment)-[:COMMENTS]->(p:Post) WHERE NOT p.deleted = true AND NOT p.disabled = true RETURN COUNT(DISTINCT(p))")
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-03-04T12:44:38Z" title="Wednesday, March 4th 2020, 1:44:38 pm +01:00">Mar 4, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -97,14 +97,14 @@ type User {
   contributions: [Post]! @relation(name: "WROTE", direction: "OUT")
   contributionsCount: Int! @cypher(
   statement: """
-  MATCH (this)-[: WROTE]->(r: Post)
+  MATCH (this)-[:WROTE]->(r:Post)
   WHERE NOT r.deleted = true AND NOT r.disabled = true
   RETURN COUNT(r)
   """
   )
 
   comments: [Comment]! @relation(name: "WROTE", direction: "OUT")
-  commentedCount: Int! @cypher(statement: "MATCH (this)-[: WROTE]->(: Comment)-[: COMMENTS]->(p: Post) WHERE NOT p.deleted = true AND NOT p.disabled = true RETURN COUNT(DISTINCT(p))")
+  commentedCount: Int! @cypher(statement: "MATCH (this)-[:WROTE]->(:Comment)-[:COMMENTS]->(p:Post) WHERE NOT p.deleted = true AND NOT p.disabled = true RETURN COUNT(DISTINCT(p))")
```


Does anybody know why we're not checking
```
WHERE NOT comment.deleted = true AND NOT comment.disabled = true
```
here?

Reply:
> [<img alt="Mogge" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-03-04T13:00:58Z" title="Wednesday, March 4th 2020, 2:00:58 pm +01:00">Mar 4, 2020</time>_

---


I understand that we count the posts commented by the user and not the comments themselves. So I think the cypher statement would be more complex. We should look for all the comments by the user that are nether disabled nor deleted and then count the posts commented by these comments.

Reply:
> [<img alt="Mogge" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-03-04T13:36:21Z" title="Wednesday, March 4th 2020, 2:36:21 pm +01:00">Mar 4, 2020</time>_

---


Does this work:
```cypher
MATCH (this)-[:WROTE]->(c:Comment)-[:COMMENTS]->(p:Post)
WHERE NOT p.deleted = true AND NOT p.disabled = true
AND NOT c.deleted = true AND NOT c.disabled = true
RETURN COUNT(DISTINCT(p))
```
What happens if a post is commented twice and one of the comments is deleted? Does it still match or is it filtered out because of the deleted comment?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-03-04T14:11:22Z" title="Wednesday, March 4th 2020, 3:11:22 pm +01:00">Mar 4, 2020</time>_

---


Exactly that would work. If you comment a post and disable the comment, that would first increase and decrease the `commentedCount`. Whereas without the extra condition, the `commentedCount` would stay the same.

Now that we're talking about it, I remember that we distinguish between `commentedCount` and `commentsCount`. One is the number of posts commented, the other is the number of comments written.

The `DISTINCT` in the query above will make sure that `commentedCount` does not increase if you comment the same post twice.

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-03-04T14:11:57Z" title="Wednesday, March 4th 2020, 3:11:57 pm +01:00">Mar 4, 2020</time>_

---


I'm going to leave this as it is though.